src/js/modules/rpc/server.ts

```diff
@@ -45,6 +55,99 @@ export class ProcessBatchServer extends SupervisorServer {
+    this.logger.info(`request enable wasm script: ${ids}`);
+    const responses = input.coprocessors.map((definition) =>
+      this.validateEnableCoprocInput(definition, this.logger)
+    );
+    return Promise.resolve({ responses });
+  }
+
+  validateEnableCoprocInput(
+    handleDef: EnableCoprocessor,
+    logger: Logger
+  ): EnableCoprocessorRequestData {
+    try {
+      const { id, source_code } = handleDef;
+      const coprocessor = this.loadCoprocFromString(id, source_code.toString());
+      // TODO: Remove Handle type.
+      const handle: Handle = {
+        coprocessor,
+        checksum: "",
+      };
+      const prevHandle = this.repository.findByCoprocessor(handle.coprocessor);
+      if (prevHandle != undefined) {
+        logger.info(`error on load wasm script: ${handleDef.id}`);
+        return errors.createResponseScriptIdAlreadyExists(prevHandle);
+      } else {
+        if (handle.coprocessor.inputTopics.length === 0) {
+          logger.info(`wasm script doesn't have topics: ${handleDef.id}`);
+          return errors.createResponseScriptWithoutTopics(handle);
+        }
+        const validTopics = handle.coprocessor.inputTopics.reduce(
+          (prev, topic) => errors.validateTopic(topic) && prev,
+          true
+        );
+        if (!validTopics) {
+          logger.info(`invalid topic on wasm script: ${handleDef.id}`);
+          return errors.createResponseScriptInvalidTopic(handle);
+        }
+        logger.info(`wasm script loaded on nodejs engine : ${handleDef.id}`);
+        this.repository.add(handle);
+        return errors.createResponseSuccess(handle);
+      }
+    } catch (e) {
+      if (e instanceof SyntaxError) {
+        logger.info(
+          `syntax error on wasm script: ${handleDef.id}, ${e.message}`
+        );
+        return errors.createResponseScriptSyntaxError(handleDef);
+      } else {
+        logger.info(`error on wasm script: ${handleDef.id}, ${e.message}`);
+        return errors.createResponseInternalError(handleDef);
+      }
+    }
+  }
+
+  loadCoprocFromString(id: bigint, script: string): Optional<Coprocessor> {
+    /**
+     * use a Function constructor, that function allows execute javascript
+     * from a string, the first and second arguments are the parameter for
+     * that function and the last one is the js string program.
+     */
+    const loadScript = Function("module", "require", script);
+    /**
+     * Create a custom type for result function, as coprocessor script return
+     * a exports.default value, ResultFunction type represents that result.
+     */
+    type ResultFunction = { exports?: { default?: Coprocessor } };
+    /**
+     * We create a 'module' result where our function save the object that
+     * coprocessor script exports.
+     */
+    const module: ResultFunction = {};
+    /**
+     * pass our module object and nodeJs require function.
+     */
+    try {
+      loadScript(module, require);
```

Comment:
Same here. If only `loadScript` can throw, then surround only that statement and leave the rest out.

Reply:
Done